README.md

```diff
@@ -47,19 +47,26 @@ To watch for changes and automatically trigger a new build:
 
 
 **Logging**
+Logging is handled by `https://github.com/winstonjs/winston`. A pre-configured logger is available, and can be used like so:
 
-By using a utility wrapper (`src/utility/logger`) surrounding `console.log`, the `awsRequestId` and a "correlation ID" is output with every debug/info/warn/error message.
+```ts
+import logger from "../utils/logger";
 
-For this pattern to work, every service/lambda must forward their correlation ID to subsequent services via a header e.g. `X-Correlation-Id`. 
+logger.info('Hello world');
+logger.error('Hello world');
+logger.warn('Hello world');
+```
 
-In practice, the first lambda invoked by an initial request will not have received the `X-Correlation-Id` header, so its `correlationId` gets defaulted to its `lambdaRequestId`.
-This `correlationId` should then be used when invoking subsequent lambdas via the `X-Correlation-Id` header.
-Every lambda called subsequently will then check for that `X-Correlation-Id` header and inject it into their logs.
+**Tip:** It's usually a good idea to attach the `requestId` and any `X-Correlation-Id` to the logger object's meta data so every log message will contain these useful bits of information.
 
-This shows an example of what the log looks like from the first invoked lambda:
-```
-2020-09-10T17:03:04.891Z	5ff37fce-5ace-114c-9120-a1406cc8d11d	INFO	{"apiRequestId":"c6af9ac6-7b61-11e6-9a41-93e8deadbeef","correlationId":"5ff37fce-5ace-114c-9120-a1406cc8d11d","message":"Here's a gnarly info message from lambda 1 - notice how my correlationId has been set to my lambdaRequestId?"}
-```
-This shows an example of what the logs look like from the second invoked lambda (called via the first lambda):
+```ts
+  const event: APIGatewayProxyEvent = currentInvoke.event as APIGatewayProxyEvent;
+  const correlationId: string = event.headers['X-Correlation-Id'] || (currentInvoke.context as Context).awsRequestId;
+
+  const { requestId } = event.requestContext;
+
+  req.app.locals.correlationId = correlationId;
+  req.app.locals.requestId = requestId;
+
+  Logger.defaultMeta = { requestId, correlationId };
```

Comment:
Will the metadata be available whenever the logger is imported into a new service? Or are you passing requestId & correlationId into each service that uses the logger?